addons/sale/models/sale.py

```diff
@@ -547,11 +547,18 @@ def _action_confirm(self):
 
     @api.multi
     def action_confirm(self):
+        if self._get_forbidden_state_confirm() & set(self.mapped('state')):
+            raise UserError(_(
+                'It is not allowed to confirm an order in the following states: %s'
+            ) % (', '.join(self._get_forbidden_state_confirm())))
```

Comment:
why you did not simply ignore it and pass? this Warning will brake automations that do not manage previously the state.

a simple ``return`` is better.

Reply:
I assumed that these transitions (done => confirmed or canceled => confirmed) is not something you would do in any case, hence the raising an error instead of simply ignoring. Moreover, a `return` would also break the automation, but silently.